pints/toy/_hes1_michaelis_menten.py

```diff
@@ -50,14 +49,75 @@ class Hes1Model(pints.ForwardModel, ToyModel):
            https://doi.org/10.1038/ncomms1496
     """
     def __init__(self, y0=None, implicit_parameters=None):
-        if y0 is None:
-            self.set_initial_conditions(2)
-        else:
-            self.set_initial_conditions(y0)
         if implicit_parameters is None:
             self.set_implicit_parameters([5., 3., 0.03])
         else:
             self.set_implicit_parameters(implicit_parameters)
+        if y0 is None:
+            self.set_initial_conditions(2)
+        else:
+            self.set_initial_conditions(y0)
+
+    def _dfdp(self, state, time, parameters):
+        """ See :meth:`pints.ToyModel.jacobian()`. """
+        m, p1, p2 = state
+        P0, v, k1, h = parameters
+        p2_over_p0 = p2 / P0
+        p2_over_p0_h = p2_over_p0**h
+        one_plus_p2_expression_sq = (1 + p2_over_p0_h)**2
+        ret = np.empty((self.n_states(), self.n_parameters()))
+        ret[0, 0] = h * p2 * p2_over_p0**(h - 1) / (
+            P0**2 * one_plus_p2_expression_sq)
+        ret[0, 1] = 0
+        ret[0, 2] = 0
+        ret[0, 3] = - (p2_over_p0_h * np.log(p2_over_p0)) / (
+            one_plus_p2_expression_sq
+        )
+        ret[1, 0] = 0
+        ret[1, 1] = m
+        ret[1, 2] = -p1
+        ret[1, 3] = 0
+        ret[2, 0] = 0
+        ret[2, 1] = 0
+        ret[2, 2] = p1
+        ret[2, 3] = 0
+        return ret
+
+    def initial_conditions(self):
+        """
+        Returns m(0).
+        """
+        return self._y0[0]
+
+    def implicit_parameters(self):
+        """
+        Returns the implicit parameters of this model.
```

Comment:
Are these from ToyODEModel?

Reply:
Yes, this is vague (and not my terminology). I may rename and will certainly re-doc!

Reply:
Thanks!

Reply:
Done